### PR TITLE
batches: hide/disable UI when user does not have permission to create batch changes

### PR DESCRIPTION
**Pull request by courier-new**
Closes https://github.com/sourcegraph/sourcegraph/issues/47219. Based on #48569.

A user who does not have the `BATCH_CHANGES#WRITE` permission will:

- See a disabled "Create" button with a tooltip reason on any variation of the batch changes list page:
    image
    image

- Not be able to access the "Create" page via URL:
    image

- See an info alert at the top of the "Getting started" page:
    image

- Not see settings related to batch changes, nor be able to access those routes:
    image
    image
	image

This PR also fixes a bug where the "Get Sourcegraph Enterprise" button would mistakenly show in place of the "Create" button when a user could not create a batch change for _any_ reason, even if their instance was enterprise and they could create batch changes in other circumstances.

## Test plan

Manual testing with a user with and without permission. 

## App preview:

- [Web](https://sg-web-kr-disable-bc.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.04% (+1.08 kb) | 0.00% (+0.25 kb) | -0.01% (-0.83 kb) | -0.13% (-1) 🔽 |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/kr/disable-bc/compare-report.html;tab=live_object) for a full comparison between the commits ebe860b and b11d927 or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

File: client/web/src/enterprise/batches/list/GettingStarted.tsx [hunk 1]
```diff
@@ -12,16 +11,22 @@ import { eventLogger } from '../../../tracking/eventLogger'
 
 export interface GettingStartedProps {
     isSourcegraphDotCom: boolean
-    authenticatedUser?: Pick<AuthenticatedUser, 'displayName' | 'emails'> | null
```
Comment (courier-new) [inline client/web/src/enterprise/batches/list/GettingStarted.tsx:15]: This prop was unused.

**Comment by danielmarquespt** [comment]
Hi!

I'm ok with hiding the sidebar setting options, but I would prefer to keep a disabled "create" button with a tooltip "Your user does not have sufficient permissions to create batch changes. Contact your site admin to request access." 

For 2 reasons:

- Avoid frustration: Lets say a user is trying to follow a tutorial or something, they won't see the button there and won't understand why

- Maybe, by explaining that this is just a matter of permissions, we might get users to ask for permission and increase usage, opposed to the user not knowing what to do to create a batch change.


For the settings pages, we can just hide the options, thats fine because thats only useful for those who already are able to create batch changes. But the main list page is improtant to display the reason why the user cannot move forward.

Wha do you think?

**Comment by eseliger** [comment]
Agree with Daniel.

**Comment by courier-new** [comment]
Makes sense to me! I initially leaned towards hiding rather than disabling because I noticed we already hide it on the list page for a specific namespace, if the user viewing that page does not have the ability to create in that specific namespace (e.g. it's the list for another user or an org they aren not part of). I think I'll update it to just make it disabled in that case, too, then.

Thanks for the feedback!! 🙇‍♀️

**Review by BolajiOlajide — APPROVED** [review]

**Pull request merged**